### PR TITLE
[CA 909] Add a test to assert service and db healthy

### PR DESCRIPTION
**Pull request by yonghaoy**

### CODE REVIEW

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        with:
+          java-version: 1.8
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
```
Comment (wchamber) [inline .github/workflows/unit_test.yml:43]: just curious, where'd you find this?

File: .github/workflows/unit_test.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Run unit tests
+        run: ./gradlew unitTest --scan
+      - name: Stop Postgres DB
+        run:  ./local-dev/run_postgres.sh stop
```
Comment (wchamber) [inline .github/workflows/unit_test.yml:49]: do we need to stop the database? Maybe keeping this just to test that `run_postgres.sh stop` works is a good reason to have it.
if so, we probably still want it to run even if the unit tests fail, as long as the initialize database step has succeeded? see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif

File: README.md [hunk 3]
```diff
@@ -44,4 +44,22 @@ You may need to use gcloud to provide GCR
 
 ```
 local-dev/setup_local_env.sh <environment>
+```
+
+## Testing
```
Comment (wchamber) [inline README.md:49]: hmm, should this go in the local_dev README instead? don't have a strong opinion.

File: src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java [hunk 4]
```diff
@@ -2,14 +2,43 @@
...
+
+  @Autowired
+  UnauthenticatedApiController(JanitorJdbcConfiguration jdbcConfiguration) {
+    this.jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+  }
+
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    return new ResponseEntity<>(new SystemStatus().ok(true), HttpStatus.OK);
+    if (jdbcTemplate.getJdbcTemplate().execute(this::isConnectionValid)) {
```
Comment (wchamber) [inline src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java:26]: wdyt about inlining `isConnectionValid`? don't think we're getting much readability benefit from having a named function.

`.execute(connection - > connection.isValid(0)) {...`

File: src/test/java/bio/terra/janitor/app/ServerStartUpTest.java [hunk 5]
```diff
@@ -0,0 +1,3 @@
+package bio.terra.janitor.app;
+
+public class ServerStartUpTest {}
```
Comment (wchamber) [inline src/test/java/bio/terra/janitor/app/ServerStartUpTest.java:3]: what's up with this file?

File: .github/workflows/unit_test.yml (hunk 1 above)
Comment (yonghaoy) [reply .github/workflows/unit_test.yml:43]: ah, I copied this from: https://github.com/DataBiosphere/terra-workspace-manager/blob/dev/.github/workflows/unit-test.yml

File: .github/workflows/unit_test.yml (hunk 2 above)
Comment (yonghaoy) [reply .github/workflows/unit_test.yml:49]: ah, sorry, forgot to delete this line. That is my previous version: Manually start and stop ad postgres in docker. 
Now we are using postgres service container (line 13-24) so we don't need to manually start and stop it.

File: README.md (hunk 3 above)
Comment (yonghaoy) [reply README.md:49]: Feels the WM way has to many readmes and hard to follow. So I am thinking about to have one readme and will have local server, deploy to gke, how to run test sections. Removed local_dev/ readme and put everythinh here.

File: src/test/java/bio/terra/janitor/app/ServerStartUpTest.java (hunk 5 above)
Comment (yonghaoy) [reply src/test/java/bio/terra/janitor/app/ServerStartUpTest.java:3]: deleted this file

File: src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java (hunk 4 above)
Comment (yonghaoy) [reply src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java:26]: done

File: .github/workflows/unit_test.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run Unit Tests
+
+on: [push]
```
Comment (wchamber) [inline .github/workflows/unit_test.yml:6]: Should this run on pull requests as well?
https://github.com/DataBiosphere/terra-cloud-resource-lib/blob/7abed537fa4a6d02189dde8b78d62aff6ebd100d/.github/workflows/build-and-test.yaml#L6-L10
